### PR TITLE
Feature: New Rails Deployment Lesson

### PR DESCRIPTION
**Pull request by KevinMulhern**
### Because:
* Heroku no longer has a free tier.

### This commit:
* Remove Heroku install step from Installing Ruby lesson - it was never needed there.
* Remove Rails "Preparing for Deployment" lesson - we no longer need it.
* Change Rails mailing lessons to refer to generic "email providers" instead of the Heroku Sendgrid add-on.
* Remove references to Heroku from the Rails final project.
* Change Rails Deployment lesson to...
  * Allow users to choose from a range of hosting providers instead of just Heroku.
  * Further explain what a hosting providers are, what a PaaS provider is and how they work.
  * More tips for troubleshooting and debugging deployment issues.

### Additional Notes:
The rationale for recommending a range of options instead of just one is because there is no like for like alternative for Heroku's free tier. All providers have a very limited free tier.

We will be advising using the free tiers of a combination of different hosting providers for free hosting. We also mention the option is there to choose one hosting provider and pay for it if the learner has the means.

Our learners come from a lot of different backgrounds and don't all have the means to pay for hosting. The different circumstances of each of our learners must be given the up-most respect by providing everyone with options so we can help as many people as possible.

### CODE REVIEW

**Comment by KevinMulhern** [comment]
Live preview of the new deployment lesson: https://www.theodinproject.com/lessons/preview?uuid=a91a1f73-0d95-4390-8345-d593c6df43d1

File: ruby_on_rails/rails_basics/deployment.md [hunk 1]
```diff
@@ -1,165 +1,226 @@
...
+In the real world, you'll want to link it to your own custom domain, for example, `http://mycooldomain.com`. It's worth mentioning that you don't need custom domain names for the portfolio projects you'll be building in this course. The random domains you're PaaS provider gives you will be good enough. But if you have an itch for tricking out your app with a custom domain name. You'll first need to purchase a domain from a registrar like [GoDaddy](https://www.godaddy.com/) or [IWantMyName](https://iwantmyname.com/).
 
-We'll do a quick overview of how it will work. It's not meant to be a step-by-step guide... for that, please check out [Heroku's "Getting Started with Rails 7.x" guide](https://devcenter.heroku.com/articles/getting-started-with-rails7).  A typical convention with Heroku commands is that they're prefixed with either `$ heroku run` or just `$heroku`, so running a database migration on Heroku is `$ heroku run rails db:migrate` and using the console is `$ heroku run console`.
+To find new a new domain, try using [Domainr](https://domainr.com/).
```
Comment (CouchofTomato) [inline ruby_on_rails/rails_basics/deployment.md:67]: duplicate 'new' here

**Review by CouchofTomato — APPROVED** [review]
I'm approving because there is only one small change to make and can be deployed right after.

File: ruby_on_rails/rails_basics/deployment.md (hunk 1 above)
Comment (KevinMulhern) [reply ruby_on_rails/rails_basics/deployment.md]: Thanks mate, great catch 💪

**Comment by Uaitt** [comment]
I'm wondering if we could add

1. [Fly.io documentation](https://fly.io/docs/)
2. [Render documentation](https://render.com/docs)
3. [Railway.app documentation](https://docs.railway.app/)

In the additional resources section.

**Comment by KevinMulhern** [comment]
Thanks for the feedback @Uaitt, I updated each option with a link to their respective docs along side the Rails guides for each service.

**Pull request merged**